### PR TITLE
WIP: ptrace for arm64

### PR DESCRIPTION
**Pull request by rminnich**
Doesn't work yet

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

### CODE REVIEW

File: pkg/strace/syscall_linux_arm64.go [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+
+package strace
+
+import (
+	"golang.org/x/sys/unix"
+)
+
+const archWidth = 64
+
+// This is the amd64 syscall map. One might think that this one map could be used for all Linux
```
Comment (rjoleary) [inline pkg/strace/syscall_linux_arm64.go:23]:
```suggestion
// This is the arm64 syscall map. One might think that this one map could be used for all Linux
```

File: pkg/strace/syscall_linux_arm64.go [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+// FillRet fills the TraceRecord with the result values from the registers.
+func (s *SyscallEvent) FillRet() {
+	s.Ret = [2]SyscallArgument{{uintptr(s.Regs.Regs[0])}}
+	if errno := int(s.Regs.Regs[0]); errno < 0 {
+		s.Errno = unix.Errno(-errno)
+	}
+}
+
+// Signal table (taken from Go runtime, but adding all-caps signal names)
+var signals = [...]string{
```
Comment (rjoleary) [inline pkg/strace/syscall_linux_arm64.go:355]: Surely this is common across all Linux?

Comment (rjoleary) [reply pkg/strace/syscall_linux_arm64.go:355]: Can reason this cannot just be `[]string`?

Comment (rminnich) [reply pkg/strace/syscall_linux_arm64.go:355]: perhaps ironically, when I first wrote this code, the internal Google reviewer demanded [...]

It's an arrray of "just the right size", at that point, not a slice.

I moved this to syscall_linux.go. I'm actually shocked that is it the same across all Linux.

Comment (hugelgupf) [reply pkg/strace/syscall_linux_arm64.go]: Y'all are misreading each other. Ryan said "surely this is" - he was asking you.

It is not, actually. See "signal numbering" in https://man7.org/linux/man-pages/man7/signal.7.html

Comment (hugelgupf) [reply pkg/strace/syscall_linux_arm64.go]: Although MIPS might be the only one that Go actually supports in that table...

**Comment by rminnich** [comment]
Someone else will have to do this one.

**Pull request closed without merging**